corehq/tests/noseplugins/elasticsearch.py

```diff
@@ -0,0 +1,13 @@
+from nose.plugins import Plugin
+
+from corehq.pillows.utils import get_all_expected_es_indices
+from corehq.util.elastic import reset_es_index
+
+
+class ElasticsearchPlugin(Plugin):
+    name = 'elasticsearch'
+    enabled = True
+
+    def configure(self, options, conf):
+        for es_info in get_all_expected_es_indices():
+            reset_es_index(es_info)
```

Comment:
How long does this take?

Assuming it's non-trivial (more than a fraction of a second): I assume this plugin will be applied even when running tests that do not touch ES, which means that if this takes even a few seconds it's overhead that will be applied to every single test run. Is there a way to apply this only when running tests that need it, perhaps with a decorator around those tests?

Also, what happens if ES is not installed? What do you think of making this conditional based on the value of `settings.SKIP_TESTS_REQUIRING_EXTRA_SETUP` or something along those lines?

Reply:
Thanks @millerdev! It apparently takes 11.9 seconds to sync elasticsearch indices (testing locally), so this is probably not how I want to do this. For the time being I'm just going to remove it from this PR entirely by reverting the commit. I need to test whether it's strictly necessary to make tests pass or if there's a manageable workaround where we do this in place.

`reset_es_index` uses `trap_extra_setup`, so `SKIP_TESTS_REQUIRING_EXTRA_SETUP` should be respected if that's called inside tests (not sure what that does this configure context, but I'll figure that out if I want to resurrect this).